src/main/java/Parser.java

```diff
@@ -0,0 +1,246 @@
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class Parser {
+    TaskList taskList;
+    Storage storage;
+
+    /**
+     * Constructor for Parser
+     * @param taskList List of tasks that the Parser interacts with
+     * @param storage Storage that the Parser interacts with
+     */
+    Parser(TaskList taskList, Storage storage) {
+        this.storage = storage;
+        this.taskList = taskList;
+    }
+
+
+
+
+    /**
+     * Update the storage after an operation is performed
+     */
+    void updateStorage() {
+        List<Task> tasks = taskList.getTaskList();
+        List<String> taskStrings = tasks.stream().map(Task :: toString).collect(Collectors.toList());
+        storage.updateFile(taskStrings);
+    }
+
+    /**
+     * Makes sense of a user's command
+     * @param command Command given
+     * @return Whether operation was successful
+     */
+    boolean handleCommand(String command) {
```

Comment:
Perhaps the name here could be changed to something that sounds more like a boolean variable? The same applies to some of the other methods in this file.